src/main/java/ratpack/zipkin/ServerTracingModule.java

```diff
@@ -15,107 +15,60 @@
   protected void configure() {
-    bind(ServerRequestAdapterFactory.class).in(SINGLETON);
-    bind(ServerResponseAdapterFactory.class).in(SINGLETON);
     bind(ServerTracingHandler.class).to(DefaultServerTracingHandler.class);
-    Provider<ServerTracingHandler> serverTracingHandlerProviderProvider = getProvider(ServerTracingHandler.class);
-
-    bind(ClientRequestAdapterFactory.class).in(SINGLETON);
-    bind(ClientResponseAdapterFactory.class).in(SINGLETON);
+    Provider<ServerTracingHandler> serverTracingHandlerProviderProvider =
+        getProvider(ServerTracingHandler.class);
 
 
     bind(HttpClient.class).annotatedWith(Zipkin.class).to(ZipkinHttpClientImpl.class);
-    Multibinder.newSetBinder(binder(), HandlerDecorator.class).addBinding().toProvider(() -> HandlerDecorator.prepend(serverTracingHandlerProviderProvider.get()));
-  }
-
-  @Provides
-  public SpanNameProvider spanNameProvider(final Config config) {
-    return config.spanNameProvider;
-  }
-
-  @Provides
-  public RequestAnnotationExtractor requestAnnotationExtractorFunc(final Config config) {
-    return config.requestAnnotationFunc;
-  }
-
-  @Provides
-  public ResponseAnnotationExtractor responseAnnotationExtractorFunc(final Config config) {
-    return config.responseAnnotationFunc;
-  }
-
-  @Provides
-  public ServerResponseInterceptor serverResponseInterceptor(final Brave brave) {
-    return new ServerResponseInterceptor(brave.serverTracer());
-  }
-
-  @Provides
-  public ServerRequestInterceptor serverRequestInterceptor(final Brave brave) {
-    return new ServerRequestInterceptor(brave.serverTracer());
-  }
 
-  @Provides
-  public ClientRequestInterceptor clientRequestInterceptor(final Brave brave) {
-    return new ClientRequestInterceptor(brave.clientTracer());
-  }
+    OptionalBinder.newOptionalBinder(binder(), SpanNameProvider.class)
+        .setDefault().to(DefaultSpanNameProvider.class).in(Scopes.SINGLETON);
 
-  @Provides
-  public ClientResponseInterceptor clientResponseInterceptor(final Brave brave) {
-    return new ClientResponseInterceptor(brave.clientTracer());
+    Multibinder.newSetBinder(binder(), HandlerDecorator.class).addBinding()
+        .toProvider(() -> HandlerDecorator.prepend(serverTracingHandlerProviderProvider.get()));
   }
 
   @Provides
-  public Brave getBrave(final Config config, final ServerConfig serverConfig) {
-    Brave.Builder braveBuilder = new Brave.Builder(
-        new RatpackServerClientLocalSpanState(
-                buildLocalEndpoint(
-                        config.serviceName,
-                        serverConfig.getPort(),
-                        serverConfig.getAddress()
-                )
-        )
-    );
-    if (config.spanReporter != null) {
-      braveBuilder.reporter(config.spanReporter);
-    }
-    else if (config.spanCollector != null) {
-      braveBuilder.spanCollector(config.spanCollector);
-    }
-    if (config.sampler != null) {
-      braveBuilder.traceSampler(config.sampler);
-    }
```

Comment:
I couldn't resist building your branch locally and giving it a spin - got an NPE due because I hadn't included a `Sampler` in my config. The way it currently is, it's optional - and I kinda like having a "batteries included" set of defaults so maybe that's something we can consider...